safe_transaction_service/history/tests/test_tx_processor.py

```diff
@@ -117,14 +123,23 @@ def test_tx_processor_with_factory(self):
         self.assertEqual(safe_status.threshold, threshold)
 
         threshold = 1
-        # Check deleting the owner did delete this pending confirmation
+        # Check deleting the owner did delete this pending confirmation for both signature and transaction
```

Comment:
nitpick: this test has around 274 lines. By adding functionality and editing the test it becomes more difficult to maintain the test.

My suggestion is to split the test into multiple ones where we test these conditions seperately. The setup might end up being the same but by adding more meaning to what is expected on each test it also makes it easier to debug.

Reply:
@fmrsabino This test checks a complete flow for indexing a Safe, and I cannot think of how to split as it depends on the status of the previous steps. This test is completely necesary and has been really useful for detecting indexing issues, but I agree that maybe we could have different tests for checking all the `tx_processor` options and branches